csharp/src/Ice/ColocatedEndpoint.cs

```diff
@@ -39,15 +39,15 @@ internal class ColocatedEndpoint : Endpoint
         public override Connection CreateDatagramServerConnection(ObjectAdapter adapter) =>
             throw new InvalidOperationException();
 
-        public override IEnumerable<Endpoint> ExpandHost(out Endpoint? publish) =>
-            throw new NotSupportedException("colocated endpoint can't be used for object adapter endpoints");
-
         public override IEnumerable<Endpoint> ExpandIfWildcard() => new Endpoint[] { this };
 
         protected internal override void AppendOptions(StringBuilder sb, char optionSeparator)
         {
         }
 
+        protected internal override Endpoint Clone(string host) =>
+            throw new NotImplementedException("cannot clone a colocated endpoint");
```

Comment:
Is `NotImplementedException` really the correct type. MS doc says:
*
>You might choose to throw a NotImplementedException exception in properties or methods in your own types when the that member is still in development and will only later be implemented in production code. In other words, a NotImplementedException exception should be synonymous with "still in development."

I think `NotSupportedException` is still the most correct here.

Reply:
Replaced by `=> this` as suggested by Benoit in an other comment.